### ISSUE TITLE
Document creation fails when \verbatiminput is used to include the main file from an inputted file in a subdirectory

### ISSUE THREAD

**Issue by hochleitner** [texstudio-org/texstudio#2670]
This is a bit of a special issue that appeared in a LaTeX template document where we demonstrate the use of `\verbatiminput` to input the contents of a whole file (e.g, source code). This verbatim input is done in a file that's included via `\input` from the main document, and we used the main document as an argument of `\verbatiminput` for demonstration purposes. Although something that's probably not done on a regular basis, it seems to create some unexpected behavior in TeXstudio, hence the issue.

### Environment
- TeXstudio: 4.3.1
- Qt: 6.3.1
- OS: Windows 10, Windows 11, Mac OS X 13
- TeX distribution: MikTex, MacTeX

### Expected behavior

Consider the following MWE:

File: `verbatiminput-mwe.tex` (main file):
```latex
\documentclass[a4paper]{article}
\usepackage{verbatim}

\begin{document}
	\include{sections/anotherfile}
\end{document}
```

File: `sections/anotherfile.tex` (located in the subdirectory "sections")
```latex
\section{I include the main file}
\verbatiminput{verbatiminput-mwe.tex}
```

The expected behavior is the creation of a document that shows the source of `verbatiminput-mwe.tex`:

```
\documentclass[a4paper]{article}
\usepackage{verbatim}

\begin{document}
	\include{sections/anotherfile}
\end{document}
```

Like this:
![image](https://user-images.githubusercontent.com/2236807/201470978-48c97c65-c0d5-4108-83db-c595320f2636.png)

### Actual behavior

What's actually happening is that TeXstudio throws these errors:

![image](https://user-images.githubusercontent.com/2236807/201471036-a0ea45a9-0ef3-48ce-b45b-300d5d2cf9ae.png)

This happens because TeXstudio actually tries to build `anotherfile.tex` and not the main file `verbatiminput.tex`:

![image](https://user-images.githubusercontent.com/2236807/201471072-ec79f549-036e-4902-9dc4-c3b245a46859.png)

Somehow TeXstudio gets thrown off by including a file and inside of this using a verbatim-input of the main file again. This is also visible in the structure view, where TeXstudio thinks that `otherfile.tex` does not exist, even though it's there:

![image](https://user-images.githubusercontent.com/2236807/201471133-adf9d1c4-724a-4a5a-a501-f69d50301c2d.png)

### How to reproduce

1. Create the main file, e.g. `verbatiminput-mwe.tex` and add the contents as shown above.
2. Create a subdirectory, e.g. `sections` and add `otherfile.tex` whit the above content.
3. Build or Build & View

On Mac OS X it seems that this construct works, als long as `otherfile.tex` is in the same directory as the main file, in a subdirectory it always fails for me though. On Windows it files no matter where the included tex file is located.

### Additional info

We've been discussing this issue in the repository of our template document here: https://github.com/Digital-Media/HagenbergThesis/issues/139

**Comment by hochleitner** [comment]
Yes, absolutely a corner case. But this problem with TeXstudio only exists for some time. We have been using this verbatiminput example in our thesis document for years and it never was an issue until recently (maybe within the last year or months).

If it helps, I can try multiple txs versions and try to track back when it started to throw an error like this.

**Comment by sunderme** [comment]
if you want to work-around, set "verbatiminput-mwe.tex" as explicit root.
